tests/utils/test_rest_util.py

```diff
@@ -454,6 +454,34 @@ def test_proxy_conditions_with_no_proxy(self):
             host = "10.0.1.1"
             self.assertEquals(should_use_proxy, use_proxy and not restutil.bypass_proxy(host))
 
+    def test_proxy_conditions_with_no_proxy_not_set(self):
+        should_use_proxy = True
+        use_proxy = True
+
+        host = "10.0.0.1"
+        self.assertEquals(should_use_proxy, use_proxy and not restutil.bypass_proxy(host))
+
+        host = "foo.com"
+        self.assertEquals(should_use_proxy, use_proxy and not restutil.bypass_proxy(host))
+
+        host = "www.google.com"
+        self.assertEquals(should_use_proxy, use_proxy and not restutil.bypass_proxy(host))
+
+        host = "168.63.129.16"
+        self.assertEquals(should_use_proxy, use_proxy and not restutil.bypass_proxy(host))
+
+        host = "www.bar.com"
+        self.assertEquals(should_use_proxy, use_proxy and not restutil.bypass_proxy(host))
+
+        host = "www.bar.com"
```

Comment:
Why is this test exactly the same as the test above? Was the flag supposed to change in the meantime?

Reply:
Cleaned the tests. I didn't need a new test. Just added new asserts to the previous test.